### ISSUE TITLE
Inspection "This statement seems to be disconnected from its parent foreach"  with objects

### ISSUE THREAD

**Issue by nickvergessen** [kalessil/phpinspectionsea#1431]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin | Php Inspections (EA Extended) v3.0.16.2 |
| Language level | PHP 7.1 |

Current behaviour
--
Code:
```php
		foreach ($participants as $participant) {
			$notification->setUser($participant['userId']);
			$this->notificationManager->notify($notification);
		}
```

![Bildschirmfoto von 2019-10-16 15-03-25](https://user-images.githubusercontent.com/213943/66922352-2f54fb00-f027-11e9-84a2-44d52eec5e36.png)

Expected behaviour
--
No warning should be generated because the object used in the function call is modified within the foreach.

**Comment by kalessil** [comment]
Thank you for reporting @nickvergessen, I'll check what we can do here.

**Comment by kalessil** [comment]
For myself: respond to https://github.com/kalessil/phpinspectionsea/issues/173#issuecomment-553714839 as well

**Comment by BenGrandin** [comment]
Here another example if it could help 

```php
		foreach ($addresses as $address) {
			[
				'_id' => $id,
				'_source' => ['adresse' => $adresse]
			] = $address;

			dump($id, $adresse);
			$this->addressConnector->deleteDocument($id, $this->newIndex);
			file_put_contents($this->deletedLogFile . '-temp', $adresse . "\n", FILE_APPEND);

		}
```

**Comment by colonelclick** [comment]
| Subject | Details |
| --- | --- |
| Plugin | Php Inspections (EA Extended) v4.0.6.4 |
| Language level | PHP 7.4 |

Here is another example using the Symfony ProgressBar object.

```
$progressBar = $this->io->createProgressBar();
$progressBar->start();

foreach ($foo as $bar) {
    ...

    //This statement seems to be disconnected from its parent foreach
    $progressBar->advance();
}
```

**Comment by tebeso** [comment]
is there a way to ignore this in phpstorm for now?

**Comment by kevinoo** [comment]
Use this to ignore the warning:
`/** @noinspection DisconnectedForeachInstructionInspection */`

Example **before the line**:

```
$progressBar = $this->io->createProgressBar();
$progressBar->start();

foreach ($foo as $bar) {
    ...

    // To ignore: "This statement seems to be disconnected from its parent foreach"
    /** @noinspection DisconnectedForeachInstructionInspection */
    $progressBar->advance();
}
```

-----

Example **before a method**:

```
class MyHappyClass 
{
    /** @noinspection DisconnectedForeachInstructionInspection */
    public function myHappyMethod(): Happiness
    {
        // Any code to make you happy
    }
}
```

-----

Example **in the file's header**:

```
<?php
/** @noinspection DisconnectedForeachInstructionInspection */

// ... Your code :-)
```

-----

Hope I was helpful :smile:
